### ISSUE TITLE
EFI rescue system automatically boots into host last saved (leaving no choice between different hosts)

### ISSUE THREAD

**Issue by guru4712** [rear/rear#2818]
#### Relax-and-Recover (ReaR) Issue Template

Fill in the following items before submitting a new issue
(quick response is not guaranteed with free support):

* ReaR version ("/usr/sbin/rear -V"): 2.6-4

* OS version ("cat /etc/os-release" or "lsb_release -a" or "cat /etc/rear/os.conf"):
RHEL 8.4

* ReaR configuration files ("cat /etc/rear/site.conf" and/or "cat /etc/rear/local.conf"):
```
OUTPUT=USB
USB_DEVICE_FILESYSTEM_LABEL='REAR-2.6.4'
USB_DEVICE=/dev/disk/by-label/$USB_DEVICE_FILESYSTEM_LABEL

AUTOSHRINK_DISK_SIZE_LIMIT_PERCENTAGE=70
AUTOINCREASE_DISK_SIZE_THRESHOLD_PERCENTAGE=200
USE_DHCLIENT=NO
USE_STATIC_NETWORKING=YES
USE_RESOLV_CONF=( NO )
PROGS+=( lsusb more )
```

* Hardware vendor/product (PC or PowerNV BareMetal or ARM) or VM (KVM guest or PowerVM LPAR):
Server Fujitsu TX1330S4

* System architecture (x86 compatible or PPC64/PPC64LE or what exact ARM device):
X86_64

* Firmware (BIOS or UEFI or Open Firmware) and bootloader (GRUB or ELILO or Petitboot):
UEFI - GRUB2

* Storage (local disk or SSD) and/or SAN (FC or iSCSI or FCoE) and/or multipath (DM or NVMe):
SSD

* Storage layout ("lsblk -ipo NAME,KNAME,PKNAME,TRAN,TYPE,FSTYPE,LABEL,SIZE,MOUNTPOINT"):
```
SYSTEM
======
/dev/sda                        /dev/sda                   disk                        223,1G
|-/dev/sda1                     /dev/sda1 /dev/sda         part vfat                     550M /boot/efi
|-/dev/sda2                     /dev/sda2 /dev/sda         part xfs                     1000M /boot
`-/dev/sda3                     /dev/sda3 /dev/sda         part LVM2_member            221,6G
  |-/dev/mapper/sina-sina--root /dev/dm-0 /dev/sda3        lvm  xfs                      206G /
  `-/dev/mapper/sina-sina--swap /dev/dm-1 /dev/sda3        lvm  swap                    15,6G [SWAP]

BOOT STICK
==========
/dev/sdb                        /dev/sdb            usb    disk                          3,7G
|-/dev/sdb1                     /dev/sdb1 /dev/sdb         part vfat        REAR-EFI     400M
`-/dev/sdb2                     /dev/sdb2 /dev/sdb         part ext3        REAR-2.6.4   3,3G
```

* Description of the issue (ideally so that others can reproduce it):

Formatted USB-stick for UEFI using above local.conf:
```
rear -D -c `pwd` format -- --efi /dev/sdb 
```
[rear-sm01-gneu_format.log](https://github.com/rear/rear/files/8853346/rear-sm01-gneu_format.log)

Prepared rescue system on first host and another one
```
rear -D -c `pwd` mkrescue
```
[rear-sm01-gneu_mkrescue.log](https://github.com/rear/rear/files/8853350/rear-sm01-gneu_mkrescue.log)
[rear-sm02-gneu_mkrescue.log](https://github.com/rear/rear/files/8853351/rear-sm02-gneu_mkrescue.log)

Booting from stick always runs automatically into the rescue system prepared by the last host saved.
But I exspected to be prompted for a choice between all the hosts saved!

This choice used to appear with version 2.5-1 on RHEL 7.6 (bios MBR).
The structure of actual ../rear/syslinux.conf is identical to that produced by version 2.5-1 mentioned above.
But behavior is different!
[syslinux.cfg.txt](https://github.com/rear/rear/files/8853357/syslinux.cfg.txt)

* Workaround, if any:

Produce ISO boot file on each server and copy on demand to usb-stick.

**Comment by pcahyna** [comment]
> This choice used to appear with version 2.5-1 on RHEL 7.6 (bios MBR).
The structure of actual ../rear/syslinux.conf is identical to that produced by version 2.5-1 mentioned above.
But behavior is different!

I think this is because with EFI, one boots using GRUB and syslinux.conf is not even used. This is unfortunate, we have other cases where the GRUB menu does not offer the same functionality as the SYSLINUX/ISOLINUX one.

**Comment by jsmeix** [comment]
On first glance this issue here seems to be
same or at least similar to item '(b)' in
https://github.com/rear/rear/issues/2666

See also
https://github.com/rear/rear/issues/2648

In current master code we have GRUB2_DEFAULT_BOOT
in default.conf which is currently online at
https://github.com/rear/rear/blob/master/usr/share/rear/conf/default.conf#L3329
but I don't know if that is also used for UEFI
and/or if it is of any use for this issue here.

**Comment by guru4712** [comment]
@jsmeix: Yes, you may regard this issue as a duplicate to item '(b)' in #2666.

And imho GRUB2_DEFAULT_BOOT seems to be of no relevance to the goal to open a "second level menu" for choosing one of multiple hosts after choosing between unsecure or secure boot from rear.

**Comment by jsmeix** [comment]
I think this issue is sufficiently answered
so I close it as "done" (we don't have a "duplicate" label)
because this issue is as duplicate of item '(b)' in
https://github.com/rear/rear/issues/2666
which is intended to get implemented for ReaR 2.8
